core/chains/solana/soltxm/txm.go

```diff
@@ -25,82 +36,341 @@ var (
+			var wg sync.WaitGroup
+			wg.Add(len(sigsBatch))
+
+			// process signatures
+			processSigs := func(s []solanaGo.Signature, res []*rpc.SignatureStatusesResult) {
+				for i := 0; i < len(res); i++ {
+					// if status is nil (sig not found), continue polling
+					// sig not found could mean invalid tx or not picked up yet
+					if res[i] == nil {
+						txm.lggr.Debugw("tx state: not found",
+							"signature", s[i],
+						)
+
+						// check confirm timeout exceeded
+						checkTimestamp(s[i], "failed to find transaction within confirm timeout")
+						continue
+					}
+
+					// if signature has an error, end polling
+					if res[i].Err != nil {
+						txm.lggr.Errorw("tx state: failed",
+							"signature", s[i],
+							"error", res[i].Err,
+							"status", res[i].ConfirmationStatus,
+						)
+						txm.txs.Revert(s[i])
+
+						// clear timestamp
+						timestampsMu.Lock()
+						delete(timestamps, s[i])
+						timestampsMu.Unlock()
+						continue
+					}
+
+					// if signature is processed, keep polling
+					if res[i].ConfirmationStatus == rpc.ConfirmationStatusProcessed {
+						txm.lggr.Debugw("tx state: processed",
+							"signature", s[i],
+						)
+
+						// check confirm timeout exceeded
+						checkTimestamp(s[i], "tx failed to move beyond 'processed' within confirm timeout")
+						continue
+					}
+
+					// if signature is confirmed/finalized, end polling
+					if res[i].ConfirmationStatus == rpc.ConfirmationStatusConfirmed || res[i].ConfirmationStatus == rpc.ConfirmationStatusFinalized {
+						txm.lggr.Debugw(fmt.Sprintf("tx state: %s", res[i].ConfirmationStatus),
+							"signature", s[i],
+						)
+						txm.txs.Success(s[i])
+
+						// clear timestamp
+						timestampsMu.Lock()
+						delete(timestamps, s[i])
+						timestampsMu.Unlock()
+						continue
+					}
+				}
+			}
+
+			// loop through batch
+			for i := 0; i < len(sigsBatch); i++ {
+				// fetch signature statuses
+				statuses, err := client.SignatureStatuses(ctx, sigsBatch[i])
+				if err != nil {
+					txm.lggr.Errorw("failed to get signature statuses in soltxm.confirm", "error", err)
+					wg.Done() // don't block if exit early
+					break     // exit switch
+				}
+
+				// nonblocking: process batches as soon as they come in
+				go func(index int) {
+					defer wg.Done()
+					processSigs(sigsBatch[index], statuses)
+				}(i)
+			}
+			wg.Wait() // wait for processing to finish
+		}
+		// TODO: defaults to 1s - should change to 0.5s
+		tick = time.After(utils.WithJitter(txm.cfg.ConfirmPollPeriod()))
+	}
+}
+
+// goroutine that simulates tx (use a bounded number of goroutines to pick from queue?)
+func (txm *Txm) simulate(ctx context.Context) {
+	defer txm.done.Done()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case msg := <-txm.simQueue:
+			// get client
+			client, err := txm.client.Get()
+			if err != nil {
+				txm.lggr.Errorw("failed to get client in soltxm.simulate", "error", err)
+				break // exit switch
```

Comment:
Would this read easier as `continue`, since then you don't have to think about the `select` at all?
```suggestion
				continue // exit switch
```